src/plugins/search-disabled-repos.py

```diff
@@ -0,0 +1,109 @@
+        suggest_enabling(conduit)
+        return
+
+    if prompt_temporarily_enable_repos(conduit):
+        conduit.info(logging.DEBUG, 'Depsolving failed due to missing dependencies, temporarily enabling repos...')
+        global attempted
+        attempted = True
+        global old_enabled_repos
+        old_enabled_repos = set((repo.id for repo in repo_storage.listEnabled()))
+        for repo in disabled_repos:
+            repo.enable()
+            conduit.info(logging.DEBUG, 'Repo temporarily enabled: %s' % repo.id)
+        repo_storage.populateSack()
+        conduit.getTsInfo().changed = True
+
+
+def postverifytrans_hook(conduit):
+    if not attempted:
+        return
+
+    used_repos = set(p.repoid for p in conduit.getTsInfo().getMembersWithState(output_states=TS_INSTALL_STATES))
+    helpful_new_repos = used_repos - old_enabled_repos
+
+    if prompt_permanently_enable_repos(conduit, helpful_new_repos):
+        for repo in helpful_new_repos:
+            # FIXME: replace with API calls
+            os.system('subscription-manager repos --enable=%s' % repo)
+            conduit.info(logging.DEBUG, 'Repo permanently enabled: %s' % repo)
+
+
+def is_repo_important(repo):
+    if repo.repofile != '/etc/yum.repos.d/redhat.repo':
+        return False
+    not_important_suffixes = ('debug-rpms', 'source-rpms', 'beta-rpms')
+    return not any(repo.id.endswith(suffix) for suffix in not_important_suffixes)
+
+
+def suggest_enabling(conduit):
+    msg = conduit.pretty_output_restring() + """\n**********************************************************************
+yum can be configured to try to resolve such errors by temporarily enabling
+disabled repos and searching for missing dependencies.
+To enable this functionality please set 'notify_only=0' in %s
+**********************************************************************\n""" % PLUGIN_CONF_PATH
+    conduit.info(1, msg) # yum's debuglevel 1
+
+
+def prompt_temporarily_enable_repos(conduit):
+    msg = conduit.pretty_output_restring() + """\n**********************************************************************
+Dependency resolving failed due to missing dependencies.
+Some repositories on your system are disabled, yum can enable them
+and search for missing dependencies. This will require downloading
+metadata for disabled repositories and may take some time and traffic.
+**********************************************************************\n"""
+    prompt = 'Enable all repositories and try again? [y/N]: '
+    return conduit.promptYN(msg, prompt=prompt)
+
+
+def prompt_permanently_enable_repos(conduit, repos):
+    msg = """*******************************************************************
+Dependency resolving was successful thanks to enabling these repositories:
+*******************************************************************\n"""
+    msg += '\n'.join(repo for repo in repos)
+    prompt = "\nWould you like to permanently enable these repositories? [y/N]: "
```

Comment:
How would you feel about putting the enabled repositories in the block marked by asterisks?  If I'm reading correctly, right now the text would be

```
*******************************************************************
Dependency resolving was successful thanks to enabling these repositories:
*******************************************************************
foobar
foobar-optional
Would you like to permanently enable these repositories? [y/N]:
```

With something like

```
msg = """*******************************************************************
Dependency resolving was successful thanks to enabling these repositories:\n%s
*******************************************************************\n"""  %
"\n".join(repos)
```

We'd get

```
*******************************************************************
Dependency resolving was successful thanks to enabling these repositories:
foobar
foobar-optional
*******************************************************************
Would you like to permanently enable these repositories? [y/N]:
```

Which sets the prompt apart from the rest of the output.